docs/README.md

```diff
@@ -431,6 +431,15 @@ If default regular expressions do not match the need, they can also be defined:
           valid-pr-title-regex: "^JKEY-[0-9]+ [A-Za-z]{1}.*$"
 ```
 
+To exempt specific branch names from the branch name checks, the optional input parameter called `whitelist-branches` can be utilized. If there are multiple branches to be excluded, they can be separated using the "|" symbol.
+
+```yaml
+      - uses: Alfresco/alfresco-build-tools/.github/actions/enforce-pr-conventions@ref
+        with:
+          jira-project-key: JKEY
+          whitelist-branches: "ABC-1234-branch|XYZ-5678-branch"
```

Comment:
I don't see why that syntax would work natively with contains, did you try this on a test branch please?
if that helps: https://stackoverflow.com/questions/75420197/how-to-use-array-input-for-a-custom-github-actions

Also there are pre-commit errors on that file

Reply:
Actually I found that we can use `contains` to check if a `current branch` is present in `whitelist-branches` or not, if present then we will not run the branch check.
More info can be found [here](https://docs.github.com/en/actions/learn-github-actions/expressions#contains).

There is one mistake that I found few minutes ago is what if someone's current branch name is `XYZ` and `whitelist-branches` is `ABC-1234-branch|XYZ-5678-branch`, `contains` will return true in this case as `whitelist-branches` has a substring with `XYZ`. So right now I am modifying it to convert to list and using for loop to check if current branch matches the lists of `whitelist-branches`

Reply:
Or shall we take json inputs as mentioned here -> https://stackoverflow.com/questions/75420197/how-to-use-array-input-for-a-custom-github-actions?

Reply:
When I look at the answer I think it's better to do that, here's a quick test that worked:

with:
```
        with:
          whitelist-branches: |-
            foo
            bar
```

some bash that did print "foo found": 
```
      run: |
        readarray -t array <<< "${{ inputs.whitelist-branches }}"
        if [[ ${array[@]} =~ 'foo' ]]
        then
          echo "foo found"
        else
          echo "foo not found"
        fi
```